stack-graphs/src/stitching.rs

```diff
@@ -746,6 +747,7 @@ pub struct ForwardPartialPathStitcher<H> {
     appended_paths: Appendables<H>,
     similar_path_detector: Option<SimilarPathDetector<PartialPath>>,
     max_work_per_phase: usize,
+    stats: Stats,
```

Comment:
Does this update mean that we'll start tracking these histograms for every stack graph operation?  Is there a worry about that affecting performance if we don't report the stats in production?  (Or are you intending to eventually expose these stats through the C API?)  Does your performance tracking bot measure _time_, or just memory usage?

Reply:
These are good questions.

>  Does this update mean that we'll start tracking these histograms for every stack graph operation? Is there a worry about that affecting performance

At the moment stats are always collected. I think the main impact would come from increased memory usage (and corresponding allocation). If you don't ask for quantiles there's not much going on.

The work in #321 might be extended to allow us to turn stats collection on and off. That would eliminate some of these concerns, I imagine.

>  Does your performance tracking bot measure time, or just memory usage?

The CI performance check doesn't seem to report a significant change in memory (y-axis) or number of operations (x-axis). But I'm not sure what number of operations says about run times.

> Or are you intending to eventually expose these stats through the C API?

I've considered that, but I don't know if that would be worth the effort?